resources/packs/azure/azure.lr

```diff
@@ -133,13 +193,64 @@ private azuread.serviceprincipal @defaults("id") {
 // Azure Resources
 azure {
   // Get all the resources in a subscription
-  resources() []azure.resource
+  resources() []azure.subscription.resource
   // Get all the resource groups in a subscription
-  resourceGroups() []azure.resourcegroup
+  resourceGroups() []azure.subscription.resourcegroup
+  // Get the subscription
+  subscription() azure.subscription
+}
+
+azure.subscription @defaults ("name") {
+  // Full resource identifier of the subscription
+  id string
+  // Name of the subscription
+  name string
+  // Subscription identifier
+  subscriptionId string
+  // Subscription tenant identifier
+  tenantId string
+  // List of tenants that manage the subscription
+  managedByTenants []string
+  // Subscription tags
+  tags map[string]string
+  // Subscription state
+  state string
+  // Subscription authorization source
+  authorizationSource string
+  // Subscription policies
+  subscriptionsPolicies dict
+  // Get resource groups inside the subscription
+  resourceGroups() []azure.subscription.resourcegroup
+  // Get compute resources inside the subscription
+  compute() azure.subscription.compute
+  // Get network resources inside the subscription
+  network() azure.subscription.network
+  // Get storage resources inside the subscription
+  storage() azure.subscription.storage
+  // Get web resources inside the subscription
+  web() azure.subscription.web
+  // Get SQL resources inside the subscription
+  sql() azure.subscription.sql
+  // Get MySQL resources inside the subscription
+  mySql() azure.subscription.mysql
+  // Get PostgreSQL resources inside the subscription
+  postgreSql() azure.subscription.postgresql
+  // Get MariaDB resources inside the subscription
+  mariaDb() azure.subscription.mariadb
+  // Get key vault resources inside the subscription
+  keyVault() azure.subscription.keyvault
+  // Get authorization resources inside the subscription
+  authorization() azure.subscription.authorization
+  // Get monitor resources inside the subscription
+  monitor() azure.subscription.monitor
+  // Get cloud defender resources inside the subscription
+  cloudDefender() azure.subscription.cloudDefender
+  // Get all the resources in a subscription
+  resources() []azure.subscription.resource
 }
 
 // Azure Resource
-private azure.resource @defaults("id name location") {
+private azure.subscription.resource @defaults("id name location") {
```

Comment:
The overlay only works if the `azure.subscription` resource has only the plural terms.